### PR TITLE
Default to styling all supported file formats

### LINKED ISSUE #963: Default to styling all supported file types in `style_dir()` and `style_pkg()`?
I am wondering if `style_dir()` and `style_pkg()` shouldn't style all supported file formats in the directory/package by default. 

Is there any reason why the users may wish to format only code in `.R` files, but not in (e.g.) `.Rmd` files? 

Current default:

```r
filetype = c("R", "Rprofile")
```

Proposed default:

```r
filetype = c("R", "Rprofile", "Rmd", "Rmarkdown", "Rnw")
```

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Closes #963

### CODE REVIEW

**Comment by IndrajeetPatil** [comment]
@lorenzwalthert All green now.

**Comment by lorenzwalthert** [comment]
Thaks for the PR 😄. A few comments:
* the `API` file should also be updated. Can you install `r-lib/pkgapi` and re-run roxygenize?
* If it is a breaking change (which I am not sure yet), we should probably not merge it into main (but like `rc-2.0.0`), fix all the other issues in #769 and then release v2.0.0.

**Comment by IndrajeetPatil** [comment]
The API file is already updated. precommit caught that.

**Comment by IndrajeetPatil** [comment]
> If it is a breaking change (which I am not sure yet)

Okay, let me know when you come to a decision on this point. It also doesn't feel like a breaking change to me, bur rather a major change.

**Comment by IndrajeetPatil** [comment]
Hmm, why did the Continuous Benchmarks GHA fail?

**Comment by lorenzwalthert** [comment]
> Why did {touchstone} fail?

@assignUser can you help here? I think it's related to some non-R code that you contributed that I can't understand. Same problems occured in https://github.com/lorenzwalthert/touchstone/issues/97#issuecomment-1162140464.

**Comment by assignUser** [comment]
The Commenting step failed because the main touchstone workflow failed and this correctly added a ❌ to the check suite. The actual touchstone run timed out during dependency installation, so probably an issue unrelated to the PR/touchstone.

So overall everything works as intended (except the time out 😁 ), I would try re-running/re-triggering the benchmark

**Comment by lorenzwalthert** [comment]
@assignUser oh yes right. Thanks for explaining. So basically every time the first part of the action fails, this same error ocurrs, right? Guess that's just something we have to know if errors occur and then dig deeper... Anyways, I restarted the benchmark. It has an old rspm checkpoint so I am surprised it failed. Let's see what happens...

**Comment by lorenzwalthert** [comment]
GitHub Actions now passing, had to update some config, see #969.

**Comment by IndrajeetPatil** [comment]
Thanks, @lorenzwalthert! Yes, the builds are indeed passing now.

Btw, having slept a bit more on it, I don't think this would qualify as a breaking change. A breaking change would have been if we had stopped supporting styling a file type. Here, we continue to support everything we were supporting _and some more file types_.

File: NEWS.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ editor_options:
 
 # styler 1.7.0.9000 (Development version)
 
+**Breaking Changes**
```
Comment (lorenzwalthert) [inline NEWS.md:9]:
```suggestion
**User-facing changes**
```

**Review by lorenzwalthert — APPROVED** [review]

**Comment by lorenzwalthert** [comment]
yeah, you are right. I marked it as *user-facing change*.

**Comment by lorenzwalthert** [comment]
Thanks 🎉

**Pull request merged**